### PR TITLE
Spring Event Day 1: Martyr Objective!

### PR DESCRIPTION
**Pull request by Cheridan**
Ordered by FreakyM!

If you have no incompatible objectives (such as Steal, etc), there will be a 1/5 chance to get the objective to go out with a bang instead of the traditional Escape.

### CODE REVIEW

**Comment by Lumipharon** [comment]
Excellent.

**Comment by optimumtact** [comment]
:-1: encourages greif and bad roleplay, also security will never be able to execute people as they might want that and we want to stop greentext wherever possible.

**Comment by Incoming5643** [comment]
For better and for worse we had this years ago

**Comment by Iamgoofball** [comment]
Yesssssss

(Fuck off @optimumtact)

**Comment by optimumtact** [comment]
Excuse me? At the very least it should be behind a config option so that server owners can determine if they want it

**Comment by kevinz000** [comment]
>die a glorious death.
Ayy someone make bombing guide pls =)

**Comment by Steelpoint** [comment]
Is there a way to tell if the user committed suicide? If so then killing yourself via suicide (Via the verb/whatever) should disqualify you.

**Comment by RemieRichards** [comment]
@Steelpoint if you have committed suicide or are in the process of committing suicide the ```suiciding```
variable is true (and remains true once you die of suicide/toolbox)

@Cheridan I agree with steelpoint, it makes sense.

**Comment by Fox-McCloud** [comment]
Don't understand why suicide should disqualify you from this, to be honest; it allows people who want to be people an easy way to off themselves without being horrendously destructive in the process of things.

There's literally no different from purchasing a syndi minibomb and holding it to gib yourself and suicide->succumbing other than forcing you to use telecrystals....likewise with beating yourself into crit then succumbing.

**Comment by RemieRichards** [comment]
Yes but @Fox-McCloud suiciding with the suicide verb is just cheap.
With a syndi bomb they're making a boom.
Beating themselves to death they leaves a corpse and a bloody item and a question of "Did he just... I mean... I guess he did"

Suiciding leaves a body with some flavour text...

And this objective is to encourage antags to do crazier/exotic stuff, suiciding, is not that.

**Comment by Fox-McCloud** [comment]
@RemieRichards 

I guess, but those who want the "cheap" victory are going to find a way anyway since there's goodness knows how many virtually zero effort ways of killing yourself in SS13.

**Comment by Nienhaus** [comment]
Well it is the "go out with a bang" and suicide isn't a bang.

**Comment by RemieRichards** [comment]
Well yes, but of the 2 you mentioned, they atleast leave something nicer than a corpse that blatantly says "I cant be arse to do something creative give me greentext"

Hell even spacing yourself leaves no evidence meaning nobody can immediately complain.

**Comment by Jordie0608** [comment]
Changelog

**Comment by Dorsisdwarf** [comment]
I love Martyr, it encourages things other than sleepypen->esword->gib->be a regular crewman for 90-120 minutes->shuttle

**Comment by demonfiren** [comment]
Suiciding with a brick of C-4 should still count, so long as it damages a handful of other people in the process.

File: code/game/gamemodes/objective.dm [hunk 1]
```diff
@@ -5,6 +5,7 @@ datum/objective
 	var/target_amount = 0				//If they are focused on a particular number. Steal objectives have their own counter.
 	var/completed = 0					//currently only used for custom objectives.
 	var/dangerrating = 0				//How hard the objective is, essentially. Used for dishing out objectives and checking overall victory.
+	var/martyr_compatible = 1			//If the objective is compatible with martyr objective, i.e. if you can still do it while dead.
```
Comment (Giacom) [inline code/game/gamemodes/objective.dm]: Shouldn't the default be 0 and then you can make the assassinate objectives set to be 1?

**Comment by Chocobro** [comment]
i'm a fan, but it toes the line between legitimate objective and grief. i'm for it, just expect there to be an outcry in OOC when the shuttle gets emagged then bombed.

**Comment by palpatine213** [comment]
My concern is that people with this objective aren't going to try to go out in some dramatic manner, but rather try to kill everyone and deny all other greentext. This is literally telling people to go murderbone. Also, sing becomes #1 greentext machine.

**Comment by Iamgoofball** [comment]
@palpatine213 "murder everyone and escape alone" isn't exactly what I'd call "glorious"

1v1 combat in the boxing ring ending with the death of both combatants is glorious

**Comment by palpatine213** [comment]
@Iamgoofball I know that, but after a while, I fear people are going to stop caring about it being "glorious" and seeing the objective as "well, I'm going to die anyway, let's see how many people I can take with me/how badly can I fuck up the station." This is just a possibility however, but I feel it might be likely.
As for glorious, are admins going to be enforcing that? Manually failing the objective? I hope it won't come to that. This can be something really cool, but has the possibility to turn into something terrible.

**Comment by Lumipharon** [comment]
These concerns are seriously overblown. If people want to blow everything up they can do it without some text suggesting something that could possibly include blowing the shit out of everything.

**Comment by MrStonedOne** [comment]
What lum said.

Hijack already causes more murderbone than this ever can, and the server gets along just fine with it.

We already have people who go by islamic sounding names who will set science to high, work toxins, every shift, than the moment they get antag, will allahu akbar the station with a bunch of bomb cap bombs.

Server gets along fine.

Every now and than we get a stretch of people who release the singularity every time they get antag.

Server gets along fine.

Coming up here soon, we will have tators getting told to do something they've already known they could do, and guess what:

Server is gonna get along just fine.

**Comment by palpatine213** [comment]
You do have a point. I retract my concern, This will be very interesting to see what scenarios will play out,

**Comment by Kromgar** [comment]
Bluespace bombign the shuttle time

**Comment by Dorsisdwarf** [comment]
merge plz?

**Comment by cookingboy3** [comment]
Sounds fun.
Allahu ackbar.

**Comment by Czarcastic** [comment]
What if they were given a free(?) suicide vest type item that could be spawned anytime with the PDA?

**Pull request merged**